### ISSUE TITLE
ASN.1 parsing failed

### ISSUE THREAD

**Issue by septs** [pyca/cryptography#10367]
```
Python 3.9.6 (default, Nov 10 2023, 13:38:27)
[Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from cryptography import x509
>>> x509.load_pem_x509_certificate(b'''-----BEGIN CERTIFICATE-----
... MIIC0zCCAnqgAwIBAgIFMzAwMDEwCgYIKoZIzj0EAwIwgaAxIDAeBgNVBAMTF0dF
... TUFMVE8gQ29ycG9yYXRlIENFIENJMQswCQYDVQQGEwJGUjEQMA4GA1UEChMHR0VN
... QUxUTzEbMBkGA1UECxMSQ29ycG9yYXRlIFNlY3VyaXR5MRAwDgYDVQQHEwdHRU1F
... Tk9TMS4wLAYJKoZIhvcNAQkBEx9jb3Jwb3JhdGUuSVRTZWN1cml0eUBnZW1hbHRv
... Y29tMCAXDTE3MDMxMzEyNTIxNloYDzIwNTIwMzEyMTI1MjE2WjCBoDEgMB4GA1UE
... AxMXR0VNQUxUTyBDb3Jwb3JhdGUgQ0UgQ0kxCzAJBgNVBAYTAkZSMRAwDgYDVQQK
... EwdHRU1BTFRPMRswGQYDVQQLExJDb3Jwb3JhdGUgU2VjdXJpdHkxEDAOBgNVBAcT
... B0dFTUVOT1MxLjAsBgkqhkiG9w0BCQETH2NvcnBvcmF0ZS5JVFNlY3VyaXR5QGdl
... bWFsdG9jb20wWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQValaTzK1xSqfxDNsG
... qDjoZkMhMBynZsquJG/ZuwZtnL3a/Wa6TCl87i196Y0vKrs28w2OH0H+vVQC9vDt
... YSxRo4GcMIGZMCAGA1UdDgEBAAQWBBR38L1p1gvcnjHacCsiHDKAi7jkKjAiBgNV
... HSMBAQAEGDAWgBR38L1p1gvcnjHacCsiHDKAi7jkKjAPBgNVHRMBAf8EBTADAQH/
... MA4GA1UdDwEB/wQEAwIBBjAXBgNVHREEEDAOiAwrBgEEAYH4AodqAQMwFwYDVR0g
... AQH/BA0wCzAJBgdngRIBAgEAMAoGCCqGSM49BAMCA0cAMEQCIFHgdppG1v76xxNV
... i0BrmPGKD/8J56Ogycw8gu3WAkQYAiAHDTeOcNcpwQysjxA7mck83zA6xGC41UVF
... URMKohsOVw==
... -----END CERTIFICATE-----''')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: error parsing asn1 value: ParseError { kind: EncodedDefault, location: ["Certificate::tbs_cert", "TbsCertificate::raw_extensions", 0, "Extension::critical"] }
```

**Comment by alex** [comment]
For ease of debugging, it's generally helpful if you paste the certificate without teh leading `...` from the Python repl.

That said, I can tell you what the problem is in this case: The rules of DER specify that when a field is encoded, if it has it's default value, that it must be omitted from the final output. In this case you have a certificate with an extension whose critical field is explicitly encoded as false, when that is the default value.

In short, whatever created this certificate is emitting invalid certs.

**Comment by septs** [comment]
https://chromium.googlesource.com/chromiumos/platform2/+/HEAD/hermes/certs/test/gemalto.pem

The certificate extracted from chromium os, it's is eSIM certificate issuer.

**Comment by alex** [comment]
I suggest someone take it up with teh issuer. There's no good reason in 2024 (or 2017, when this appears to have been issued), to be creating semantically invalid DER certificates.
